src/acceptance-test/java/de/zalando/aruha/nakadi/webservice/VersionControllerAT.java

```diff
@@ -0,0 +1,74 @@
+package de.zalando.aruha.nakadi.webservice;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import static com.jayway.restassured.RestAssured.given;
+import de.zalando.aruha.nakadi.config.JsonConfig;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.http.HttpStatus;
+import static org.hamcrest.CoreMatchers.is;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.Test;
+import org.springframework.util.StringUtils;
+
+public class VersionControllerAT extends BaseAT {
+    private static final String VERSION_RESOURCE = "/version";
+    private static final ObjectMapper mapper = (new JsonConfig()).jacksonObjectMapper();
+
+    @Test(timeout = 10000)
+    public void scmSourceShouldContainCorrectData() {
+        given().when().get(VERSION_RESOURCE).then().statusCode(HttpStatus.SC_OK).body(is(new Matcher<Object>() {
+            String error = null;
+
+            @Override
+            public boolean matches(final Object item) {
+                try {
+                    final Map info = mapper.readValue(
+                            new ByteArrayInputStream(item.toString().getBytes()), Map.class);
+                    if (null == info) {
+                        error = "Failed to read data from version " + item;
+                        return false;
+                    } else {
+                        final Map scmSource = (Map) info.get("scm_source");
+                        if (null == scmSource) {
+                            error = "Scm source is not specified in " + item;
+                            return false;
+                        }
+                        for (final String key : new String[]{"revision", "url", "author"}) {
+                            final String value = (String) scmSource.get(key);
+                            if (StringUtils.isEmpty(value)) {
+                                error = "field scm_source." + key + " is " + value + ", but content expected";
+                                return false;
+                            }
+                        }
+                        if (scmSource.get("status") == null) {
+                            error = "scm_source.status field in null";
+                            return false;
+                        }
+                        return true;
+                    }
+                } catch (final IOException e) {
+                    error = "Failed to read versionInfo from json: " + e.getMessage();
+                    return false;
+                }
+            }
+
+            @Override
+            public void describeMismatch(Object item, Description mismatchDescription) {
+                mismatchDescription.appendText("Error: " + error + " for " + item);
+            }
+
+            @Override
+            public void _dont_implement_Matcher___instead_extend_BaseMatcher_() {
```

Comment:
I think this method speaks for itself :) You should extend BaseMatcher, not implement Matcher.